apps/orders/views.py

```diff
@@ -98,5 +111,9 @@ def item_remove(self, request, *args, **kwargs):
         validated_data = serializer.validated_data
 
         user_cart = self.request.user.get_user_cart()
-        user_cart.remove_item(validated_data['product_id'])
+        deleted, _ = user_cart.items.filter(product=validated_data['product']).delete()
```

Comment:
tu ridici not found deja dupa ce s-a facut request in baza, si reiese ca request-ul s-a facut pe degeaba. 
verifica inainte sa faci request in baza